store/app/routers/orders.py

```
@@ -77,13 +80,24 @@ async def get_user_orders_with_products(
         orders = await crud.get_orders_by_user_id(user.id)
         orders_with_products = []
         for order in orders:
-            if order.stripe_product_id is None:
-                continue  # Skip orders without a stripe_product_id
-            product = await stripe.get_product(order.stripe_product_id, crud)
-            orders_with_products.append(OrderWithProduct(order=order, product=ProductInfo(**product)))
+            try:
+                if order.stripe_product_id is None:
+                    continue
+                product = await stripe.get_product(order.stripe_product_id, crud)
+                orders_with_products.append(OrderWithProduct(order=order, product=ProductInfo(**product)))
+            except Exception as e:
+                logger.error(
+                    "Error processing order", extra={"order_id": order.id, "error": str(e), "user_id": user.id}
+                )
+                continue
         return orders_with_products
     except ItemNotFoundError:
-        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="No orders found for this user")
+        return []
+    except Exception as e:
+        logger.error("Error fetching user orders", extra={"user_id": user.id, "error": str(e)})
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=f"Error fetching orders: {str(e)}"
+        )
```

exception wrapping is way too broad here. as a general principle we should wrap the part of the endpoint that we expect to have an exception, so that we are not swallowing exceptions that could tell us something meaningful